src/backend/joanie/core/serializers.py

```diff
@@ -165,6 +165,49 @@ def get_target_courses(self, product):
             context=context,
         ).data
 
+    def get_orders(self, instance):
+        """
+        If a user is authenticated retrieves valid or pending orders related to the
+        product instance. If a course code has been provided through query
+        parameters order are also filtered by course.
+        """
+        try:
+            filters = {"owner__username": self.context["username"]}
+            if course_code := self.context.get("course_code"):
+                filters.update({"course__code": course_code})
+
+            orders = models.Order.objects.filter(
+                is_canceled=False,
+                product=instance,
+                **filters,
+            ).only("pk")
+
+            return [order.pk for order in orders]
+        except (KeyError, models.Order.DoesNotExist):
```

Comment:
In my opinion this is a bad practice. You should try/except KeyError around the precise clause and try/except DoesNotExist around the get. Who knows if something in the filter call is not going to through a KeyError that should be raised!

Reply:
This bad habbit sticks to me.